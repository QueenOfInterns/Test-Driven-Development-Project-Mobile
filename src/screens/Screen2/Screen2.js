/* eslint-disable react-native/no-inline-styles */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React, {Component} from 'react';
import {View, Text, FlatList} from 'react-native';

export class Screen2 extends Component {
  constructor(props) {
    super(props);

    if (props.data) {
      this.state = {
        users: [
          {name: 'Spongebob', phoneNumber: '479-387-5534'},
          {name: 'Patrick', phoneNumber: '479-256-5344'},
          {name: 'Sandy', phoneNumber: '367-899-7070'},
          {name: 'Squidward', phoneNumber: '367-345-9999'},
          {name: 'Mr. Krabs', phoneNumber: '504-214-3322'},
          {name: 'Gary', phoneNumber: '601-405-9977'},
        ],
        error: false,
      };
    } else {
      this.state = {
        users: [],
        error: true,
      };
    }
  }
  renderInfo = ({item}) => {
    return (
      <View
        style={{
          justifyContent: 'center',
          flex: 1,
          alignItems: 'center',
          padding: 50,
        }}>
        <Text style={{padding: 20, flex: 1}}>{item.name}</Text>
        <Text>{item.phoneNumber}</Text>
      </View>
    );
  };

  render() {
    return (
      <View>
        <FlatList
          keyExtractor={item => item.id}
          data={this.state.users}
          renderItem={this.renderInfo}
        />
      </View>
    );
  }
}
