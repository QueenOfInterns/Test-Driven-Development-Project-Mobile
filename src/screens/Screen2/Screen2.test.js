/* eslint-disable no-undef */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {Screen2} from './Screen2.js';

describe('Screen 2', () => {
  describe('Flatlist', () => {
    it('can render with the given data', () => {
      const props = {
        data: [{title: 'TestTitle', body: 'TestBody'}],
        error: false,
      };
<<<<<<< HEAD
      const renderer = new ShallowRenderer();
      renderer.render(<Screen2 {...props} />);
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('does not crash with no data', () => {
      const props = {
        data: [],
        error: false,
      };
=======
>>>>>>> origin/dev-colby

      const renderer = new ShallowRenderer();
      renderer.render(<Screen2 {...props} />);
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
});
