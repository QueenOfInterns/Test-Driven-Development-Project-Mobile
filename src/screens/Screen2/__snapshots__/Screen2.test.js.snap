// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen 2 Flatlist can render with the given data 1`] = `
<View>
  <FlatList
    data={
      Array [
        Object {
          "name": "Spongebob",
          "phoneNumber": "479-387-5534",
        },
        Object {
          "name": "Patrick",
          "phoneNumber": "479-256-5344",
        },
        Object {
          "name": "Sandy",
          "phoneNumber": "367-899-7070",
        },
        Object {
          "name": "Squidward",
          "phoneNumber": "367-345-9999",
        },
        Object {
          "name": "Mr. Krabs",
          "phoneNumber": "504-214-3322",
        },
        Object {
          "name": "Gary",
          "phoneNumber": "601-405-9977",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;

exports[`Screen 2 Flatlist displays error when there is an error 1`] = `
<View>
  <FlatList
    data={Array []}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;

exports[`Screen 2 Flatlist does not crash with no data 1`] = `
<View>
  <FlatList
    data={
      Array [
        Object {
          "name": "Spongebob",
          "phoneNumber": "479-387-5534",
        },
        Object {
          "name": "Patrick",
          "phoneNumber": "479-256-5344",
        },
        Object {
          "name": "Sandy",
          "phoneNumber": "367-899-7070",
        },
        Object {
          "name": "Squidward",
          "phoneNumber": "367-345-9999",
        },
        Object {
          "name": "Mr. Krabs",
          "phoneNumber": "504-214-3322",
        },
        Object {
          "name": "Gary",
          "phoneNumber": "601-405-9977",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
